src/python/pants/backend/openapi/goals/tailor.py

```diff
@@ -0,0 +1,84 @@
+# Copyright 2022 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+from __future__ import annotations
+
+import itertools
+from dataclasses import dataclass
+from typing import Iterable
+
+from pants.backend.openapi.dependency_inference import OpenApiDependencies, ParseOpenApiSources
+from pants.backend.openapi.target_types import (
+    OPENAPI_FILE_EXTENSIONS,
+    OpenApiDefinitionGeneratorTarget,
+    OpenApiSourceGeneratorTarget,
+)
+from pants.core.goals.tailor import (
+    AllOwnedSources,
+    PutativeTarget,
+    PutativeTargets,
+    PutativeTargetsRequest,
+    group_by_dir,
+)
+from pants.engine.fs import PathGlobs, Paths
+from pants.engine.internals.native_engine import Digest
+from pants.engine.internals.selectors import Get
+from pants.engine.rules import Rule, collect_rules, rule
+from pants.engine.unions import UnionRule
+from pants.util.logging import LogLevel
+
+
+@dataclass(frozen=True)
+class PutativeOpenApiTargetsRequest(PutativeTargetsRequest):
+    pass
+
+
+@rule(level=LogLevel.DEBUG, desc="Determine candidate OpenAPI targets to create")
+async def find_putative_targets(
+    req: PutativeOpenApiTargetsRequest,
+    all_owned_sources: AllOwnedSources,
+) -> PutativeTargets:
```

Comment:
it is convenient to provide with an option in the subsystem that allows to disable tailor for this specific backend, as in the following link:
https://github.com/pantsbuild/pants/blob/bba429d42ed6521cb378ef0d069e3bfd1f4994fc/src/python/pants/backend/helm/goals/tailor.py#L35